src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -548,7 +578,13 @@ internal void StartOperationAccept()
                 }
             }
 
-            if (SocketsTelemetry.Log.IsEnabled()) SocketsTelemetry.Log.AcceptStart(_currentSocket!._rightEndPoint!);
+            if (SocketsTelemetry.Log.IsEnabled())
+            {
+                SocketsTelemetry.Log.AcceptStart(_currentSocket!._rightEndPoint!);
+
+                // Ignore _flowExecutionContext when using Telemetry to avoid losing Activity tracking
+                _context = ExecutionContext.Capture();
```

Comment:
I'd prefer we not spread out the places we call Capture.  We should instead change wherever we're skipping the setting of _context to factor in IsEnabled there.

Reply:
We have call sites like:
```c#
e.StartOperationCommon(...);
e.StartOperationConnect(multipleConnect, ...);
```

We capture the context in `StartOperationCommon`, before we decide whether to log the Start event.

I will change it so that we always log the Start before calling into `StartOperationCommon` and avoid capturing the context twice. SAEA misuse that would cause `StartOperationCommon` to throw should be rare anyway.

Reply:
Changed.